File: screen_recorder.cpp
// screen_recorder.cpp
#include <iostream>
#include <cstdlib>
#include <string>
#include <sstream>
#include <chrono>
#include <thread>

#ifdef _WIN32
#include <windows.h>
#else
#include <unistd.h>
#endif

using namespace std;

bool checkFFmpeg() {
    int ret = system("ffmpeg -version > nul 2>&1");
    return ret == 0;
}

string getFFmpegCmd(const string& output, int duration) {
#ifdef _WIN32
    return "ffmpeg -y -f gdigrab -i desktop -t " + to_string(duration) + " -c:v libx264 -pix_fmt yuv420p " + output;
#elif __APPLE__
    return "ffmpeg -y -f avfoundation -i 1 -t " + to_string(duration) + " -c:v libx264 -pix_fmt yuv420p " + output;
#else
    return "ffmpeg -y -f x11grab -i :0.0 -t " + to_string(duration) + " -c:v libx264 -pix_fmt yuv420p " + output;
#endif
}

int main(int argc, char* argv[]) {
    string output = "screen_recording.mp4";
    int duration = 10;
    for (int i = 1; i < argc; ++i) {
        string arg = argv[i];
        if (arg == "-o" && i+1 < argc) {
            output = argv[++i];
        } else if (arg == "-t" && i+1 < argc) {
            duration = stoi(argv[++i]);
        } else if (arg == "-h" || arg == "--help") {
            cout << "Usage: screen_recorder [-o output] [-t seconds]" << endl;
            return 0;
        }
    }
    if (!checkFFmpeg()) {
        cerr << "Ошибка: FFmpeg не найден. Установите FFmpeg и добавьте в PATH." << endl;
        return 1;
    }
    string cmd = getFFmpegCmd(output, duration);
    cout << "Начинаем запись на " << duration << " секунд в " << output << "..." << endl;
    int ret = system(cmd.c_str());
    if (ret != 0) {
        cerr << "❌ Ошибка записи: код " << ret << endl;
        return 1;
    }
    cout << "✅ Запись завершена. Файл: " << output << endl;
    return 0;
}

File: screen_recorder.java
// screen_recorder.java
import java.io.*;
import java.util.*;

public class screen_recorder {
    private static boolean checkFFmpeg() {
        try {
            ProcessBuilder pb = new ProcessBuilder("ffmpeg", "-version");
            pb.redirectErrorStream(true);
            Process p = pb.start();
            p.waitFor();
            return p.exitValue() == 0;
        } catch (Exception e) {
            return false;
        }
    }

    private static String getFFmpegCmd(String output, int duration) {
        String os = System.getProperty("os.name").toLowerCase();
        if (os.contains("win")) {
            return "ffmpeg -y -f gdigrab -i desktop -t " + duration + " -c:v libx264 -pix_fmt yuv420p " + output;
        } else if (os.contains("mac")) {
            return "ffmpeg -y -f avfoundation -i 1 -t " + duration + " -c:v libx264 -pix_fmt yuv420p " + output;
        } else {
            return "ffmpeg -y -f x11grab -i :0.0 -t " + duration + " -c:v libx264 -pix_fmt yuv420p " + output;
        }
    }

    public static void main(String[] args) throws IOException, InterruptedException {
        String output = "screen_recording.mp4";
        int duration = 10;
        for (int i = 0; i < args.length; i++) {
            if (args[i].equals("-o") && i+1 < args.length) {
                output = args[++i];
            } else if (args[i].equals("-t") && i+1 < args.length) {
                duration = Integer.parseInt(args[++i]);
            } else if (args[i].equals("-h") || args[i].equals("--help")) {
                System.out.println("Usage: java screen_recorder [-o output] [-t seconds]");
                return;
            }
        }
        if (!checkFFmpeg()) {
            System.err.println("Ошибка: FFmpeg не найден. Установите FFmpeg и добавьте в PATH.");
            System.exit(1);
        }
        String cmd = getFFmpegCmd(output, duration);
        System.out.println("Начинаем запись на " + duration + " секунд в " + output + "...");
        ProcessBuilder pb = new ProcessBuilder(cmd.split(" "));
        pb.redirectErrorStream(true);
        pb.redirectOutput(ProcessBuilder.Redirect.INHERIT);
        Process p = pb.start();
        int code = p.waitFor();
        if (code != 0) {
            System.err.println("❌ Ошибка записи: код " + code);
            System.exit(1);
        }
        System.out.println("✅ Запись завершена. Файл: " + output);
    }
}
